File: Java_Basic_Jan_2023_SolutionsOfProblems/P03Ex_ConditionalStatements_Advanced/P04_FishingBoat.java
package P03Ex_ConditionalStatements_Advanced;

import java.util.Scanner;

public class P04_FishingBoat {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int budget = Integer.parseInt(scanner.nextLine());
        String season = scanner.nextLine();
        int numberFishers = Integer.parseInt(scanner.nextLine());

        double price = 0;

        switch (season) {
            case "Spring":
                price = 3000;
                break;
            case "Summer":
                price = 4200;
                break;
            case "Autumn":
                price = 4200;
                break;
            case "Winter":
                price = 2600;
                break;
        }
        if (numberFishers <= 6) {
            price = price * 0.9;
        } else if (numberFishers <= 11) {
            price = price * 0.85;
        } else {
            price = price * 0.75;
        }

        if (numberFishers % 2 == 0 && (!season.equals("Autumn"))) {
            price = price * 0.95;
        }

        if (budget < price) {
            System.out.printf("Not enough money! You need %.2f leva.", price - budget);
        } else {
            System.out.printf("Yes! You have %.2f leva left.", budget - price);
        }
    }
}

File: JavaScript_Basic_04_2023_SolutionsOfProblems/03.Ex_PB-JS-Nested-Conditional-Statements/P04_FishingBoat.js
function fishingBoat(input) {
    let budget = Number(input[0]);
    let season = input[1];
    let fishermans = Number(input[2]);

    let boatCost = 0.0;

    switch (season) {
        case "Spring":
            boatCost = 3000;
            if (fishermans % 2 === 0) {
                boatCost = boatCost * 0.95;
            }
            break;
        case "Summer":
            boatCost = 4200;
            if (fishermans % 2 === 0) {
                boatCost = boatCost * 0.95;
            }
            break;
        case "Autumn":
            boatCost = 4200;
            break;
        case "Winter":
            boatCost = 2600;
            if (fishermans % 2 === 0) {
                boatCost = boatCost * 0.95;
            }
            break;
    }
    if (fishermans >= 12) {
        boatCost = boatCost * 0.75;
    } else if (fishermans >= 7) {
        boatCost = boatCost * 0.85;
    } else {
        boatCost = boatCost * 0.9;
    }
    
    if (boatCost > budget) {
        console.log(`Not enough money! You need ${(boatCost - budget).toFixed(2)} leva.`);
    } else {
        console.log(`Yes! You have ${(budget - boatCost).toFixed(2)} leva left.`);
    }
}

fishingBoat(["3600",
"Autumn",
"6"])



